js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetChecks.tsx

```diff
@@ -117,6 +123,17 @@ export const AssetChecks = ({
     return null;
   }
 
+  if (data.assetNodeOrError.__typename === 'AssetNode') {
+    switch (data.assetNodeOrError.assetChecksOrError.__typename) {
+      case 'AssetCheckNeedsAgentUpgradeError':
+        return <AgentUpgradeRequired />;
+      case 'AssetCheckNeedsMigrationError':
+        return <MigrationRequired />;
+      case 'AssetCheckNeedsUserCodeUpgrade':
+        return <NeedsUserCodeUpgrade />;
+    }
+  }
+
   if (!checks.length || !selectedCheck || !assetNode) {
```

Comment:
this feels a little fragile. If we miss one of the error types above, then along with https://github.com/dagster-io/dagster/pull/20317/files#diff-2ffee45e31514bcd6d0ac70c9ff24b98847a6c9dd3e9201213b97c62aed85ef1R83 we'll fallback to displaying `No checks defined for this asset` which is misleading. Maybe a log at least if it's not in the switch case?

Reply:
I can add an assertion to make sure we checked all of them

Reply:
Done